lib/dialects/sqlite/query-generator.js

```diff
@@ -79,6 +81,153 @@ const QueryGenerator = {
     return !!value ? 1 : 0;
   },
 
+  // A recursive parser for nested where conditions
+  // (copied from ../postgres/query-generator.js)
+  parseConditionObject(_conditions, path) {
+    path = path || [];
+    return Utils._.reduce(_conditions, (r, v, k) => { // result, key, value
+      if (Utils._.isObject(v)) {
+        r = r.concat(this.parseConditionObject(v, path.concat(k))); // Recursively parse objects
+      } else {
+        r.push({ path: path.concat(k), value: v });
+      }
+      return r;
+    }, []);
+  },
+
+  handleSequelizeMethod(smth, tableName, factory, options, prepend) {
+    if (smth instanceof Utils.Json) {
+      // Parse nested object
+      if (smth.conditions) {
+        const conditions = _.map(this.parseConditionObject(smth.conditions), condition =>
+          `json_extract(${this.quoteIdentifier(_.first(condition.path))}, '$.${_.tail(condition.path).join('.')}') = '${condition.value}'`
+        );
+
+        return conditions.join(' and ');
+      } else if (smth.path) {
+        let str;
+
+        // Allow specifying conditions using the sqlite json1 functions
+        // https://sqlite.org/json1.html
+        if (/(json(?:_.*)*)\(.*\)/i.test(smth.path)) {
+          str = smth.path;
+        } else {
+          // Also support json dot notation
+          const path = smth.path.split('.');
+          str = `json_extract(${this.quoteIdentifier(_.first(path))}, '$.${_.tail(path).join('.')}')`;
+        }
+
+        if (smth.value) {
+          str += util.format(' = %s', this.escape(smth.value));
+        }
+
+        return str;
+      }
+    } else {
+      return MySqlQueryGenerator.handleSequelizeMethod.call(this, smth, tableName, factory, options, prepend);
```

Comment:
I dont want you to move it, just use replace `MySqlQueryGenerator => AbstractQueryGenerator`